src/OAuth2User.ts

```diff
@@ -39,13 +45,21 @@ export class OAuth2User implements OAuthClient {
   code_verifier?: string;
   code_challenge?: string;
   private _refreshAccessTokenPromise: Promise<{token: Token}> | null;
+  private _tokenEvents = new EventEmitter();
   constructor(options: OAuth2UserOptions) {
     const { token, ...defaultOptions } = options;
     this.options = {client_secret: '', ...defaultOptions};
     this.token = token;
     this._refreshAccessTokenPromise = null;
   }
 
+    /**
+   * Subscribe to the events
+   */
+  async on(eventName: EventName, listener: Listener): Promise<void> {
```

Comment:
Is there an `off` method? (not sure it is needed?)

Reply:
do we nee that?. client can just remove the code for ```.on```  if they don't want to listen to the token changes and process it

Reply:
How do they remove the code of .on?

Reply:
I mean they just don't listen using on method. cause the callback is fixed here. which passes the token or an error

Reply:
they will not call the on method if they don't want to listen

Reply:
cc @bumi wdyt? or shall we implement it? just in case for future reference?

Reply:
resolved. we don't need this currently